### PR TITLE
Add admin APIs to force-join users to groups and manage their flair

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/matrix-org/synapse/issues/9143

Though the groups API is disappearing soon, these functions are intended to make flair management easier in the short term.

*This is for FOSDEM.*

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

**Comment by turt2live** [comment]
Typically I would try to write tests for this stuff, but given groups are awful and need to be burned at the stake I'm inclined to avoid doing so.

File: synapse/handlers/groups_local.py [hunk 1]
```diff
@@ -365,6 +365,32 @@ async def join_group(self, group_id, user_id, content):
...
+        """
+        if not self.is_mine_id(group_id):
+            raise SynapseError(400, "Can only affect local groups")
+
+        if not self.is_mine_id(user_id):
+            raise SynapseError(400, "Can only affect local users")
+
+        # Bypass the group server to avoid business logic regarding whether or not
+        # the user can actually join.
+        await self.store.add_user_to_group(group_id, user_id)
```
Comment (erikjohnston) [inline synapse/handlers/groups_local.py:379]: In some ways this belongs in `groups_server.py`, but its probably fine to ignore the split between the two files in this particular instance.

**Comment by clokep** [comment]
I believe we decided this can be closed?

**Comment by turt2live** [comment]
yea, the feature is being obliterated. I'd like to keep the branch around just in case though.

**Pull request closed without merging**